Automattic/wp-calypso
# Calypso Build: Remove calypso-color-schemes dep (alternate lyrics)

#### Changes proposed in this Pull Request

Alternative to @jsnajdr's #36471, implementing ideas from https://github.com/Automattic/wp-calypso/pull/36471#issuecomment-545991575 (and below).

The basic idea for this PR is to have `calypso-build` pick up a project's own `postcss.config.js` (if there is one, much like we already do for Webpack and Babel). This allows us to drop the `calypso-color-schemes` dependency from the default configuration, but add it back e.g. to Calypso-the-app (without having to resort to Webpack config/"preset"-level configs).

From the Changelog:

```
- Remove `@automattic/calypso-color-schemes` dependency
  - See the example given at `https://github.com/Automattic/wp-calypso/blob/master/packages/calypso-build/README.md#advanced-usage-use-own-postcss-config` for instructions on how to continue
    to use color definitions from that file in your project.
- Remove `calypso-color-schemes` Sass Prelude.
  - Remove the `calypso-color-schemes` colors prelude from the default webpack config.
    This was useful when the colors were defined as SCSS variables, but now when they are
    defined as CSS variables included in the CSS output, the prelude no longer makes sense.
- Use Consumer's `postcss.config.js`, if present.
- Remove `preserveCssCustomProperties` option.
```

#### Testing instructions

- Verify that Calypso still builds, runs, and looks the same.
- For bonus points, `npm link` a different project (e.g. Jetpack) to this branch's `calypso-build`, and verify that it also still builds/runs/looks the same. You might have to include a custom `postcss.config.js`, as described in the updated README.

Fixes #36466.
